source/fundamentals/crud/read-operations/count.txt

```diff
@@ -0,0 +1,256 @@
+       | Default: ``nil``
+
+   * - ``Skip`` 
+     - | The number of documents to skip before counting.
+       | Default: ``0``
+
+.. tip::
+
+   When you use ``CountDocuments()`` to return the total number of documents in a
+   collection, MongoDB performs a collection scan. You can avoid a collection scan and
+   improve the performance of this method by using a hint to take advantage of the built-in index on
+   the ``_id`` field. Use this technique only when calling ``CountDocuments()``
+   with an empty query parameter.
+
+   .. code-block:: csharp
+      :emphasize-lines: 1, 3
+
+      var filter = Builders<Student>.Filter.Empty;
+
+      CountOptions opts = new CountOptions(){Hint = "_id_"};
+      var counter = collection.CountDocuments(filter, opts);
+
+.. _csharp-estimated-count:
+
+Estimated Count
+---------------
+
+To estimate the number of documents in your collection without the use
+of a filter, use the
+``EstimatedDocumentCount()`` method. 
+
+.. note:: 
+
+    The ``EstimatedDocumentCount()`` method is quicker than the
+    ``CountDocuments()`` method because it uses the collection's
+    metadata rather than scanning the entire collection. 
+
+Modify Behavior
+~~~~~~~~~~~~~~~
+
+You can modify the behavior of ``EstimatedDocumentCount()`` by passing
+in an ``EstimatedDocumentCountOptions`` type. If you don't specify any
+options, the driver uses its default values.
+
+The ``EstimatedDocumentCountOptions`` type allows you to configure
+options with the following property:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``MaxTime``
+     - | The maximum amount of time that the query can run on the server.
+       | Default: ``nil``
+
+Example
+```````
+
+The following example estimates the number of documents in the
+``grades`` collection:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var count = collection.EstimatedDocumentCount();
+      String output = String.Format("Estimated number of documents in
+      the grades collection: " + count);
+      Console.WriteLine(output);
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Estimated number of documents in the grades collection: 6
+
+.. _csharp-count-aggregation:
+
+Aggregation
+-----------
+
+You can also include the :manual:`$count </reference/operator/aggregation/count/>`
+stage to count the number of documents in an aggregation pipeline.
+
+Example
+~~~~~~~
+
+The following example performs the following actions:
+
+- Counts the number of documents where the ``finalGrade`` is greater than ``80``
+- Assigns the count to the ``count`` field
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var matchStage = Builders<Student>
+      .Filter.Gt(f => f.finalGrade, 80);
+      var pipeline = new EmptyPipelineDefinition<Student>()
+      .Match(matchStage).Count();
+      Console.WriteLine(pipeline);
```

Comment:
I: format so that newlined method calls are indented. Also, this code should print the result of the aggregation. LMK if you want to review this code together.